### DISCUSSION TITLE
Subscription example using graphql-ws has typescript error

### DISCUSSION THREAD

**Discussion by Yuripetusko** [urql-graphql/urql#3121] [Q&A] [answered]
Using `urql@4.0.0` and `graphql-ws@5.12.1` as per latest documentation throws typescript error on optional `query` file of `FetchBody`

```
Types of property 'query' are incompatible.     Type 'string | undefined' is not assignable to type 'string'.       Type 'undefined' is not assignable to type 'string'.
```

<img width="1157" alt="image" src="https://user-images.githubusercontent.com/666169/229237798-c31931be-4f3d-4900-bb4c-f8fb2fef6828.png">

**Comment by kitten** [comment, marked as answer]
You have three options here basically, either you could:
- cast `request` to `as SubscribePayload`
- recreate the input object and annotate `request.query!` as non-optional
- just throw a random error if `!request.query`

To explain why `FetchBody.query` is optional; basically, when you use persisted queries, e.g. using `@urql/exchange-persisted` and persisted queries become “active” then `request.query` becomes `undefined` when the `query` is left out and instead a persisted hash is sent.

If you don't use the `subscriptionExchange` for either queries or mutations then it's safe to do any of the three things above regardless, since `FetchBody.query` can only be `undefined` for queries, unless the options of the `persistedExchange` have been changed, if you even use that exchange

**Reply by Yuripetusko** [reply]
Thanks, will work with that
